molo/core/wagtail_hooks.py

```diff
@@ -73,7 +73,10 @@ def copy_translation_pages_hook(request, page, new_page):
 
 @hooks.register('before_delete_page')
 def delete_page_translations(request, page):
-    if request.method == 'POST' and page.specific.language.is_main_language:
+    # If this is the main language page then we want to delete all translations
+    # but Main and Index pages don't have a language or translations
+    if request.method == 'POST' and hasattr(page.specific, 'language') and \
```

Comment:
`hasattr(page.specific, 'language')` => `hasattr(page.specific, 'language', None)`

Reply:
@moh-moola do you mean i should change the whole statement to this ?
`if request.method == 'POST' and getattr(page.specific, 'language', None).is_main_language:`
because then it will throw the error that `None` doesn't have `is_main_language`

Reply:
@KaitCrawford totally didn't think that through... you are right